src/main/scala/org/broadinstitute/dsde/firecloud/MethodsClient.scala

```diff
@@ -0,0 +1,67 @@
+package org.broadinstitute.dsde.firecloud
+
+import akka.actor.{Actor, ActorRef, Props}
+import akka.event.Logging
+import org.broadinstitute.dsde.firecloud.MethodsClient.MethodsListRequest
+import org.broadinstitute.dsde.firecloud.model.MethodEntity
+import org.broadinstitute.dsde.firecloud.model.MethodEntityJsonProtocol._
+import spray.client.pipelining._
+import spray.http.HttpHeaders.Cookie
+import spray.http.{HttpRequest, RequestProcessingException, StatusCodes}
+import spray.httpx.SprayJsonSupport._
+import spray.json.DefaultJsonProtocol
+import spray.routing.RequestContext
+
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
+
+object MethodsClient {
+  
+  case class MethodsListRequest()
+
+  def props(requestContext: RequestContext): Props = Props(new MethodsClient(requestContext))
+
+}
+
+class MethodsClient(requestContext: RequestContext) extends Actor {
+
+  import DefaultJsonProtocol._
+  import system.dispatcher
+
+  implicit val system = context.system
+  val log = Logging(system, getClass)
+
+  override def receive: Receive = {
+    case MethodsListRequest => listMethods(sender())
+  }
+
+  def listMethods(senderRef: ActorRef): Unit = {
+    log.debug("Find all methods in the methods repository")
+
+    val pipeline: HttpRequest => Future[List[MethodEntity]] =
+      (addHeader(Cookie(requestContext.request.cookies))
+        ~> sendReceive
+        ~> unmarshal[List[MethodEntity]])
+
+    val responseFuture: Future[List[MethodEntity]] = pipeline {
+      Get(FireCloudConfig.Methods.baseUrl)
+    }
+
+    responseFuture onComplete {
+      case Success(methods) =>
+        log.debug("methods found")
+        requestContext.complete(methods)
+        context.stop(self)
+
+      case Failure(error) =>
```

Comment:
this case clause catches an http error from Agora, then rethrows an http 500. Instead of catching and rethrowing, can we just bubble up the Agora error? Or at minimum, call failWith the same error code that Agora gave us? (this was an issue in vault-api too)